sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/uninstall/UninstallResourceCleaner.java

```diff
@@ -42,8 +44,9 @@
      */
     @Override
     public Collection<Resource> getPersistentVolumesToBeDestroyed(Offer offer) {
-        List<Resource> resources = offer.getResourcesList().stream().
-                filter(resource -> ResourceUtils.getPersistenceId(resource) != null).collect(Collectors.toList());
+        List<Resource> resources = offer.getResourcesList().stream()
+                .filter(resource -> resource.hasDisk() && resource.getDisk().hasPersistence())
```

Comment:
nitpick: I'm a little worried about us now having this pattern of checking persistence getting copy pasted around.

Can it just become a public method of Resource?